source/common/router/config_impl.cc

```diff
@@ -65,6 +65,24 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHost& vhost, const Json::Obj
   if (!(isRedirect() ^ !cluster_name_.empty())) {
     throw EnvoyException("routes must be either redirects or cluster targets");
   }
+
+  if (route.hasObject("headers")) {
+    std::vector<Json::Object> config_headers = route.getObjectArray("headers");
+    for (const Json::Object& header_map : config_headers) {
+      HeaderData header_data;
+
+      if (!header_map.hasObject("header_name")) {
+        throw EnvoyException("headers in the route config must have a header name");
```

Comment:
don't bother with this check here. Just let JsonException be thrown if value does not exist